src/language/mixins.md

```diff
@@ -89,6 +89,50 @@ A `mixin` declaration defines a mixin. A `class` declaration defines a [class][]
 A `mixin class` declaration defines a class that is usable as both a regular class
 and a mixin, with the same name and the same type.
 
+Any restrictions that apply to classes and mixins also apply to mixin classes:
+
+- Mixins can't have `extends` or `with` clauses, so neither can a `mixin class`.
+- Classes can't have an `on` clause, so neither can a `mixin class`. 
+
+### `abstract mixin class`
+
+You can achieve similar behavior to the `on` directive for a mixin class. 
```

Comment:
`on` clauses are really confusing. Fortunately, almost no one actually needs to use them.

People often think they are used to specify the methods than a mixin can call on itself and then get implemented by the thing that applies the mixin. That's what your example is about here.

But you don't need an `on` clause for that at all. As you do here, you can just make the type abstract and define abstract methods. Another way to accomplish something similar is to put an `implements` clause on the mixin or mixin class and then not actually implement the interface.

The only time you need an `on` clause is when the mixin wants to do `super` calls. It specifies the interface that *those* are resolved against. So in:

```dart
mixin M {}

class B {}
class C extends B with M {}
```

If there's some method in C that you want to be able to call from M, you can just declare it abstract in M and call it, like in your example:

```dart
mixin M {
  void c(); // Declare abstract.
  void test() {
    c();
  }
}

class B {}
class C extends B with M {
  void c() { ... }
}
```

You need an `on` clause if M wants to have a `super` call that needs to be resolved against *B* in the example here:

```dart
abstract class BStuff {
  void b();
}

mixin M on BInterface {
  void test() {
    super.b();
  }
}

class B {
  void b() { ... }
}

class C extends B with M {}
```

Honestly, `on` clauses and `super` calls in mixins are just not a great feature. Almost any time you think you need one, there's probably a better design you should use instead. But Flutter used `super` calls in a couple of key mixins early on (because the VM allowed it even though the language didn't specify it), so we were sort of forced to turn it into a real feature. But I definitely wouldn't emphasize it in the tour since it tends to confuse users and it's rarely what they want.

Reply:
Summarizing: I think your example is OK here, but I wouldn't mention it in the context of `on` clauses. If that makes the entire example not worth keeping, I think it's fine to eliminate this section.

Reply:
The text before this section contains these sentences:

> Sometimes you might want to restrict the types that can use a mixin.
> For example, the mixin might depend on being able to invoke a method
> that the mixin doesn't define.

That is confusing given what you just said. There might not be time for this but it might make sense to reorder this page i.e. have an example of calling the mixing classes method from the mixin near the top (where the `on` example currently is), then have a section about `class mixin` and then finally a section explaining `on` with the "you will almost certainly never need" caveat and an explanation that it is only useful for `super` calls.

Reply:
Hmm.. ok, I think what I'll do _for now_ is keep the new section referencing similarity to using `on`, but more of an aside the purpose statement it currently is. Like Brian said, this is a direct response to the earlier text describing `on`. 

Later* I'll rewrite the section to take out any emphasis on `on` and instead describe the same example for both mixins and mixin classes simply as "the use case where you want to require that whatever uses the mixin specifies some methods".

*Only delaying it because, again like Brian said, I think it'd be best to rearrange the page, since ...
- this isn't just a use case for mixin classes, 
- and, like Bob said, the existing description of `on` for regular mixins seems misrepresented and could probably use a rewrite

And doing that will take a bit of attention that I won't have before publishing the new pages. I'd like to include this info on the first publish. @munificent if you think the content just absolutely shouldn't be published, please let me know!

Reply:
My two cents: It's worth mentioning in passing for coverage. It should be reviewed in the next week or so to see if it's truly relevant. I'm a fan of being opinionated in docs, even if the opinion is to skip something entirely.